kms/keys.go

```diff
@@ -0,0 +1,236 @@
+package kms
+
+import (
+	"go.uber.org/zap"
+	"google.golang.org/api/cloudkms/v1"
+)
+
+type CryptoKey struct {
+	ID               uint `gorm:"primarykey"`
+	ProjectID        string
+	CreateTime       string
+	Name             string
+	NextRotationTime string
+
+	PrimaryAlgorithm string
+
+	CaviumCerts          []*CryptoKeyCertificateChainsCaviumCerts          `gorm:"constraint:OnDelete:CASCADE;foreignKey:ID"`
+	GoogleCardCerts      []*CryptoKeyCertificateChainsGoogleCardCerts      `gorm:"constraint:OnDelete:CASCADE;foreignKey:ID"`
+	GooglePartitionCerts []*CryptoKeyCertificateChainsGooglePartitionCerts `gorm:"constraint:OnDelete:CASCADE;foreignKey:ID"`
+
+	PrimaryAttestationContent string
+	PrimaryAttestationFormat  string
+
+	PrimaryCreateTime       string
+	PrimaryDestroyEventTime string
+	PrimaryDestroyTime      string
+
+	PrimaryExternalProtectionLevelOptionsExternalKeyUri string
+
+	PrimaryGenerateTime        string
+	PrimaryImportFailureReason string
+	PrimaryImportJob           string
+	PrimaryImportTime          string
+	PrimaryName                string
+	PrimaryProtectionLevel     string
+	PrimaryState               string
+
+	Purpose        string
+	RotationPeriod string
+
+	VersionTemplateAlgorithm       string
+	VersionTemplateProtectionLevel string
+}
+
+func (CryptoKey) TableName() string {
+	return "gcp_kms_crypto_keys"
+}
+
+type CryptoKeyCertificateChainsCaviumCerts struct {
+	ID                           uint `gorm:"primarykey"`
+	CryptoKeyCertificateChainsID uint
```

Comment:
This should be `CryptoKeyID` (ParentStruct + `ID`). see https://gorm.io/docs/has_many.html

Reply:
This comment applies to all other keys.